ConfigurationSystem/Agent/GOCDB2CSAgent.py

```diff
@@ -0,0 +1,247 @@
+# $HeadURL$
+"""
+Module provides GOCDB2CSAgent functionality.
+"""
+
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.Core.LCG.GOCDBClient import GOCDBClient
+from DIRAC.Core.Utilities.SitesDIRACGOCDBmapping import getDIRACGOCDictionary
+from DIRAC.ConfigurationSystem.Client.Helpers.Path import cfgPath
+from DIRAC.ConfigurationSystem.Client.CSAPI import CSAPI
+from DIRAC.ConfigurationSystem.Client.Config import gConfig
+
+__RCSID__ = "$Id: $"
+
+class GOCDB2CSAgent ( AgentModule ):
+  """
+  Class to retrieve information about service endpoints
+  from GOCDB and update configuration stored by CS
+  """
+
+  def initialize( self ):
+
+    # client to connect to GOCDB
+    self.GOCDBClient = GOCDBClient()
+
+    # API needed to update configuration stored by CS
+    self.csAPI = CSAPI()
+    return self.csAPI.initialize()
+
+  def execute( self ):
+    '''
+    Execute GOCDB queries according to the function map
+    and user request (options in configuration).
+    '''
+
+    # __functionMap is at the end of the class definition
+    for option, functionCall in GOCDB2CSAgent.__functionMap.iteritems():
+      optionValue = self.am_getOption( option, True )
+      if optionValue:
+        result = functionCall( self )
+        if not result['OK']:
+          gLogger.error( "%s() failed with message: %s" % ( functionCall.__name__, result['Message'] ) )
+        else:
+          gLogger.info( "Successfully executed %s" % functionCall.__name__ )
+
+    return S_OK()
+
+  def updatePerfSONARConfiguration( self ):
+    '''
+    Get current status of perfSONAR endpoints from GOCDB
+    and update CS configuration accordingly.
+    '''
+    __functionName = '[updatePerfSONAREndpoints]'
```

Comment:
The log messages prepended with a common string should rather be coded as :
log = self.log.getSubLogger( 'updatePerfSONAREndpoints' )
...
log.debug( 'Blah...' )